### PR TITLE
[doc](arrow-flight-sql) Add arrow flight sql docs

### PR DESCRIPTION
**Pull request by xinyiZzz**
since 2.1

### CODE REVIEW

**Comment by shasevendow** [comment]
Please also add the file in sidebars.json & version-2.1-sidebars.json & version-2.1.json

**Comment by xinyiZzz** [comment]
> Please also add the file in sidebars.json & version-2.1-sidebars.json & version-2.1.json

done, version-2.1.json seem not need modify

**Review by shasevendow — APPROVED** [review]
LGTM

**Review by luzhijing — APPROVED** [review]
Some wrong links

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+
+Doris implements high-speed data links based on the Arrow Flight SQL protocol, and supports multiple languages ​​to use SQL to read large batches of data from Doris at high speed.
+
+## Usage
+
+To load large batches of data from Doris to other components, such as Python/Java/Spark/Flink, you can use ADBC/JDBC based on Arrow Flight SQL to replace the past JDBC/Pymysql/Pandas to obtain higher reading performance, which is often encountered in scenarios such as data science and data lake analysis.
+
+Apache Arrow Flight SQL is a protocol developed by the Apache Arrow community to interact with database systems. It is used for ADBC ​​clients to interact with databases that implement the Arrow Flight SQL protocol using the Arrow data format. It has the speed advantage of Arrow Flight and the ease of use of JDBC/ODBC.
+
+The motivation, design and implementation, performance test results, and more concepts about Arrow Flight and ADBC ​​for Doris to support Arrow Flight SQL can be found at: https://github.com/apache/doris/issues/25514. This document mainly introduces the use of Doris Arrow Flight SQL and some common problems.
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:39]: ": https://github.com/apache/doris/issues/25514" -> [GitHub Issue](https://github.com/apache/doris/issues/25514)

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+    queryResult.close();
+    stmt.close();
+}  catch (Exception e) {
+    e.printStackTrace();
+}
+```
+
+### Choice of Jdbc and Java connection methods
+
+Compared with the traditional `jdbc:mysql` connection method, the performance test of the Arrow Flight SQL connection method of Jdbc and Java can be found at https://github.com/apache/doris/issues/25514. Here are some usage suggestions based on the test conclusions.
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:430]: Use markdown link format.

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+4. When reading a large amount of data, Arrow Flight SQL will use less memory than `jdbc:mysql://`, so if you are troubled by insufficient memory, you can also try Arrow Flight SQL.
+
+5. In addition to the above three connection methods, you can also use the native FlightClient to connect to Arrow Flight Server, which can read multiple endpoints in parallel more flexibly. Flight AdbcDriver is also a link created based on FlightClient, which is simpler than using FlightClient directly.
+
+## Interaction with other big data components
+
+### Spark & ​​Flink
+
+Arrow Flight currently has no official plan to support Spark and Flink (https://github.com/apache/arrow-adbc/issues/1490). Doris' own Spark Connector (https://github.com/apache/doris-spark-connector) and Flink Connector (https://github.com/apache/doris-flink-connector) do not currently support accessing Doris through Arrow Flight SQL. Among them, Doris Flink Connector supporting Arrow Flight SQL is under development, and it is expected to improve the reading performance several times.
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:446]: Links need to fix.

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+
+The community previously referred to the open source Spark-Flight-Connector (https://github.com/qwshen/spark-flight-connector) and used FlightClient in Spark to connect to Doris for testing. It was found that the data format conversion between Arrow and Doris Block is faster, which is 10 times the conversion speed between CSV format and Doris Block, and it has better support for complex types such as Map and Array. This is because the Arrow data format has a high compression rate and low network overhead during transmission. However, Doris Arrow Flight has not yet implemented multi-node parallel reading. It still aggregates query results to a BE node and returns them. For simple batch export of data, the performance may not be as fast as Doris Spark Connector, which supports Tablet-level parallel reading. If you want to use Arrow Flight SQL to connect to Doris in Spark, you can refer to the open source https://github.com/qwshen/spark-flight-connector and https://github.com/dremio-hub/dremio-flight-connector to implement it yourself.
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:448]: Links need to fix.

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+
+The Linux kernel version of kylinv10 SP2 and SP3 is only up to 4.19.90-24.4.v2101.ky10.aarch64. You cannot continue to upgrade the kernel version. You can only recompile Doris BE on kylinv10. If the problem still exists after compiling Doris BE with the new version of ldb_toolchain, you can try to compile it with the lower version of ldb_toolchain v0.17. If your ARM environment cannot connect to the Internet, Huawei Cloud provides ARM + kylinv10, and Alibaba Cloud provides x86 + kylinv10
+
+2. Currently, `jdbc:arrow-flight-sql` and Java ADBC/JDBCDriver do not support prepared statement passing parameters. For example, `select * from xxx where id=?` will report an error `parameter ordinal 1 out of range`. This is a bug in Arrow Flight SQL (https://github.com/apache/arrow/issues/40118)
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:458]: Links need to fix.

File: docs/db-connect/arrow-flight-sql-connect.md [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+
+3. Modification `jdbc:arrow-flight-sql` The batch size of each read can improve performance in some scenarios. By modifying the `setTargetBatchSize` in the `makeJdbcConfig` method in the `org.apache.arrow.adbc.driver.jdbc.JdbcArrowReader` file, the default is 1024, and then saving the modified file to the local directory with the same name, it will overwrite the original file to take effect.
+
+4. ADBC ​​v0.10, JDBC and Java ADBC/JDBCDriver do not support parallel reading, and the `stmt.executePartitioned()` method is not implemented. You can only use the native FlightClient to implement parallel reading of multiple Endpoints, using the method `sqlClient=new FlightSqlClient, execute=sqlClient.execute(sql), endpoints=execute.getEndpoints(), for(FlightEndpoint endpoint: endpoints)`. In addition, the default AdbcStatement of ADBC ​​V0.10 is actually JdbcStatement. After executeQuery, the row-format JDBC ResultSet is converted back to the Arrow column-format. It is expected that Java ADBC ​​will be fully functional by ADBC ​​1.0.0 https://github.com/apache/arrow-adbc/issues/1490.
```
Comment (luzhijing) [inline docs/db-connect/arrow-flight-sql-connect.md:462]: Links need to fix.

File: docs/db-connect/arrow-flight-sql-connect.md (hunk 1 above)
Comment (xinyiZzz) [reply docs/db-connect/arrow-flight-sql-connect.md:39]: fixed

**Pull request merged**